server/pkg/user/members.go

```diff
@@ -13,50 +13,74 @@ var (
 )
 
 type Members struct {
-	members map[ID]Role
-	fixed   bool
+	users        map[ID]Role
+	integrations map[IntegrationID]Role
+	fixed        bool
 }
 
 func NewMembers() *Members {
-	m := &Members{members: map[ID]Role{}}
+	m := &Members{
+		users:        map[ID]Role{},
+		integrations: map[IntegrationID]Role{},
+	}
 	return m
 }
 
 func NewFixedMembers(u ID) *Members {
-	m := &Members{members: map[ID]Role{u: RoleOwner}, fixed: true}
+	m := &Members{
+		users:        map[ID]Role{u: RoleOwner},
+		integrations: map[IntegrationID]Role{},
+		fixed:        true,
+	}
 	return m
 }
 
-func NewMembersWith(members map[ID]Role) *Members {
-	m := &Members{members: map[ID]Role{}}
-	for k, v := range members {
-		m.members[k] = v
+func NewMembersWith(users map[ID]Role) *Members {
+	m := NewMembers()
+	for k, v := range users {
+		m.users[k] = v
 	}
 	return m
 }
 
-func NewFixedMembersWith(members map[ID]Role) *Members {
-	m := &Members{members: map[ID]Role{}, fixed: true}
-	for k, v := range members {
-		m.members[k] = v
+func NewFixedMembersWith(users map[ID]Role) *Members {
+	m := &Members{
+		users:        map[ID]Role{},
+		integrations: map[IntegrationID]Role{},
```

Comment:
```suggestion
		users: maps.Clone(users),
```

Please ensure that map is initialized in all constructors or add a process for when map is nil in each method. Right now some methods do not initialize users or integrations.